File: emp_backend/src/main/java/com/example/demo/model/Employee.java
package com.example.demo.model;
import java.time.LocalDate;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.Column;
import lombok.*;


@Getter
@Setter
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Entity(name = "employees")
public class Employee{
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private long id;
	private String firstname;
	private String lastname;
	@Column(unique = true)
	private String email;
	private Long salary;
	private String department;
	private String designation;
	private LocalDate joiningDate;
}

File: emp_frontend/src/app/models/employee.ts
import {User} from "./user";

export class Employee extends User{
  firstname: string;
  lastname: string;
  salary: number;
  department: string;
  designation: string;
  joiningDate: string; // ISO 8601 format (e.g., "2024-07-18")



  constructor() {
    super();
    this.firstname = "";
    this.lastname = "";
    this.salary = 0;
    this.department = "";
    this.designation = "";
    this.joiningDate = "";
  }

}
